### ISSUE TITLE
Build failure: bcompare

### ISSUE THREAD

**Issue by ckt114** [NixOS/nixpkgs#239096]
### Steps To Reproduce

1. build *bcompare*

### Build log

```
warning: Git tree '/home/ctran/.dotfiles' is dirty
building the system configuration...
warning: Git tree '/home/ctran/.dotfiles' is dirty
error: hash mismatch in fixed-output derivation '/nix/store/gfnbcn9xs5bnydmpwy87007cp8in1xgp-bcompare-4.4.4.27058_amd64.deb.drv':
         specified: sha256-8wJzCCfekr/mrDJCDgoIqMRz21wWjfp5c5sPavZma3g=
            got:    sha256-4e0cKzGwLdsvAvqee/H1BKm5Br5Sjgt8S1Z+6vXYJzU=
error: 1 dependencies of derivation '/nix/store/abfssvm6xr6q8cl1z01qbbifkhsapyqg-bcompare-4.4.4.27058.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rpd4w4i1xdsbcc0ssml6pczqff2p00j7-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ix8vrlm38nalwxyamxq2qnlz85jyxh25-nixos-system-t490-23.05.20230619.ad157fe.drv' failed to build
```

### Notify maintainers
@ktor @arkivm  

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.01 MiB download, 0.05 MiB unpacked):
  /nix/store/37p8gq9zijbw6pj3lpi1ckqiv18j2g62-stdenv-linux
copying path '/nix/store/37p8gq9zijbw6pj3lpi1ckqiv18j2g62-stdenv-linux' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.31, NixOS, 23.05 (Stoat), 23.05.20230619.ad157fe`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by kirillrdy** [comment]
should be fixed by https://github.com/NixOS/nixpkgs/pull/232504

**Comment by ckt114** [comment]
Thanks, @kirillrdy.

**Comment by raspher** [comment]
should be fixed on 23.11 by https://github.com/NixOS/nixpkgs/commit/227188b7b304648a30faff33db57d2896703fc50, 23.05 is deprecated - please upgrade. If the problem still persists, please reopen
